### PR TITLE
JAVACLIENT-107: add isProxy property to Document class

### PR DESCRIPTION
**Pull request by funshodavid**

### CODE REVIEW

File: nuxeo-java-client/src/main/java/org/nuxeo/client/api/objects/Document.java [hunk 1]
```diff
@@ -101,6 +107,30 @@ public Document(String id, String type, List<String> facets, String changeToken,
         this.dirtyProperties = new HashMap<>();
     }
 
+    /**
+     * @since 2.3
+     */
+    public Document(String id, String type, List<String> facets, String changeToken, String path, String state,
+            String lockOwner, String lockCreated, String repositoryName, String versionLabel, String isCheckedOut,
+            Map<String, Object> properties, Map<String, Object> contextParameters, String isProxy) {
```
Comment (troger) [inline nuxeo-java-client/src/main/java/org/nuxeo/client/api/objects/Document.java]: I think it's better to have the `isProxy` argument after the `isCheckedOut` one to ease the readability. Moreover, `isProxy` should be a `boolean` (or `Boolean` if really needed?).

File: nuxeo-java-client-test/src/test/java/org/nuxeo/client/test/marshallers/DocumentMarshaller.java [hunk 2]
```diff
@@ -91,10 +92,12 @@ protected static Document readDocument(JsonParser jp) throws IOException {
                 readProperties(jp, properties);
             } else if ("changeToken".equals(key)) {
                 changeToken = jp.getText();
+            } else if ("isProxy".equals(key)) {
+                isProxy = jp.getText();
```
Comment (troger) [inline nuxeo-java-client-test/src/test/java/org/nuxeo/client/test/marshallers/DocumentMarshaller.java]: Why not getting the `isProxy` field directly as a boolean? We know it must be present, and that's a boolean. (yes it's not consistent with the other boolean fields but we can start here :)

File: nuxeo-java-client/src/main/java/org/nuxeo/client/api/objects/Document.java [hunk 3]
```diff
@@ -81,6 +81,12 @@
 
     protected String name;
 
+    /**
+     * @since 2.3
+     */
+    protected String isProxy;
+
+
     public Document(String id, String type, List<String> facets, String changeToken, String path, String state,
```
Comment (troger) [inline nuxeo-java-client/src/main/java/org/nuxeo/client/api/objects/Document.java]: This constructor should delegate to the new one, passing `null` for `isProxy`.
It should also probably be deprecated as it's not used anymore in our code, and the `isProxy` field should be there for this client version as it's only compatible with LTS 2016+.

File: nuxeo-java-client-test/src/test/java/org/nuxeo/client/test/marshallers/DocumentMarshaller.java [hunk 4]
```diff
@@ -91,10 +92,12 @@ protected static Document readDocument(JsonParser jp) throws IOException {
                 readProperties(jp, properties);
             } else if ("changeToken".equals(key)) {
                 changeToken = jp.getText();
+            } else if ("isProxy".equals(key)) {
+                isProxy = jp.getBooleanValue();
```
Comment (troger) [inline nuxeo-java-client-test/src/test/java/org/nuxeo/client/test/marshallers/DocumentMarshaller.java]: Seems that `jp.getBooleanValue()` returns a `boolean` so you can simplify to use `boolean` everywhere.

File: nuxeo-java-client/src/main/java/org/nuxeo/client/api/objects/Document.java [hunk 5]
```diff
@@ -81,9 +81,27 @@
 
     protected String name;
 
+    /**
+     * @since 2.3
+     */
+    protected Boolean isProxy;
```
Comment (troger) [inline nuxeo-java-client/src/main/java/org/nuxeo/client/api/objects/Document.java]: `boolean`

File: nuxeo-java-client/src/main/java/org/nuxeo/client/api/objects/Document.java [hunk 6]
```diff
@@ -81,9 +81,27 @@
...
+
+    /**
+     * @deprecated since 2.3
+     */
     public Document(String id, String type, List<String> facets, String changeToken, String path, String state,
```
Comment (troger) [inline nuxeo-java-client/src/main/java/org/nuxeo/client/api/objects/Document.java]: Missing `@Deprecated` annotation

File: nuxeo-java-client/src/main/java/org/nuxeo/client/api/objects/Document.java [hunk 7]
```diff
@@ -81,9 +81,27 @@
...
             String lockOwner, String lockCreated, String repositoryName, String versionLabel, String isCheckedOut,
             Map<String, Object> properties, Map<String, Object> contextParameters) {
+        this(id, type, facets, changeToken, path, state, lockOwner, lockCreated, repositoryName, versionLabel,
+                isCheckedOut, null, properties, contextParameters);
```
Comment (troger) [inline nuxeo-java-client/src/main/java/org/nuxeo/client/api/objects/Document.java]: Guess we can pass `false` if there is no `isProxy` argument as it won't be the case anymore on LTS 2016+

**Comment by kevinleturc** [comment]
👍

**Pull request merged**